modules/shared/src/main/scala/org/tessellation/schema/trust.scala

```diff
@@ -1,13 +1,113 @@
 package org.tessellation.schema
 
+import cats.syntax.either._
+
+import scala.util.control.NoStackTrace
+
 import org.tessellation.schema.peer.PeerId
 
-import derevo.cats.show
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import io.estatico.newtype.macros.newtype
 
 object trust {
 
+  object TrustValueOperations {
+
+    protected[trust] def addDoubles(
+      operand1: Option[Double],
+      operand2: Option[Double]
+    ): Either[ValueArithmeticError, Option[Double]] = {
+      val sum = operand1.flatMap(x => operand2.map(x + _)).get
+
+      if (sum >= -1.0 && sum <= 1.0) {
+        Option[Double](sum).asRight[ValueArithmeticError]
+      } else ValueOverflow.asLeft[Option[Double]]
```

Comment:
I suggest you to use refined types for that:

https://github.com/fthomas/refined

You can then create a type like:

```scala
type TrustValue = Double Refined Interval.Closed[-1.0, 1.0]
```

and then use built-in `refineV` function which returns you `Either[String, TrustValue]`.

The nice thing if refined types is that it works at compile-time for refined types. For example such thing won't compile:


```scala
val x: TrustValue = 1.1 // Compilation error because it should be <= 1.0
```


Actually it applies to all the computations below :)

Reply:
Added refined types.